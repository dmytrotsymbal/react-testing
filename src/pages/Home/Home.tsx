type Props = {}
const Home = (props: Props) => {
    const arr = ['html', 'css', 'js']

    return (
        <div className="HomePage" data-testid="homePage">
            <h1 className="HomeTitle">UnitTest_1</h1>

            <ul>
                {arr.map((item) => (
                    <li style={{ color: 'red' }} key={item}>
                        {item}
                    </li>
                ))}
            </ul>
        </div>
    )
}
export default Home
