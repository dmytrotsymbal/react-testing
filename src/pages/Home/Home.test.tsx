import { render, screen } from '@testing-library/react'
import Home from './Home'
import '@testing-library/jest-dom/extend-expect'

test('home snapshot', () => {
    const { asFragment } = render(<Home />)
    expect(asFragment()).toMatchSnapshot()
})

//_____________________________________________________________________________________

describe('Elements render', () => {
    test('renders UnitTest_1 title', () => {
        render(<Home />)
        const titleElement = screen.getByRole('heading')
        expect(titleElement).toBeInTheDocument()
    })

    test('renders list ul', () => {
        render(<Home />)
        const ulElement = screen.getByRole('list')
        expect(ulElement).toBeInTheDocument()
    })
})

//_____________________________________________________________________________________

describe('List contains', () => {
    test('list contains html', () => {
        render(<Home />)
        const listElement = screen.getByText(/html/i)
        expect(listElement).toBeInTheDocument()
    })
    test('list contains css', () => {
        render(<Home />)
        const listElement = screen.getByText(/css/i)
        expect(listElement).toBeInTheDocument()
        expect(listElement).toHaveStyle({
            color: 'red',
        })
    })
    test('list contains js', () => {
        render(<Home />)
        const listElement = screen.getByText(/js/i)
        expect(listElement).toBeInTheDocument()
    })
})

//_____________________________________________________________________________________
