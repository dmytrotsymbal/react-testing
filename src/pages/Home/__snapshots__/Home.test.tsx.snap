// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home snapshot 1`] = `
<DocumentFragment>
  <div
    class="HomePage"
    data-testid="homePage"
  >
    <h1
      class="HomeTitle"
    >
      UnitTest_1
    </h1>
    <ul>
      <li
        style="color: red;"
      >
        html
      </li>
      <li
        style="color: red;"
      >
        css
      </li>
      <li
        style="color: red;"
      >
        js
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
